src/kzg.rs

```diff
@@ -85,11 +86,12 @@ mod tests {
 
     #[test]
     fn kzg_test() {
-        let p_committed = Polynomial::create_polynomial(7);
+        //Create a polynomial with given degree.
+        let polynomial_degree = 7;
```

Comment:
You can carry this inside the prover because no one needs to know polynomial degree. It is a data leak